src/replay_stage.rs

```diff
@@ -152,6 +153,7 @@ impl ReplayStage {
                     cluster_info.write().unwrap().set_leader(scheduled_leader);
                 }
 
+                //todo what about this? if this is removed, leader blocks outside of its slot are being sent
```

Comment:
@carllin @rob-solana, I've mostly narrowed down my issues to this line. 
What do I do about this. I seem to have much better behavior without this line entirely but I fail these tests 
```
---- fullnode::tests::test_validator_to_leader_transition stdout ----
thread 'fullnode::tests::test_validator_to_leader_transition' panicked at 'assertion failed: `(left == right)`
  left: `39`,
 right: `32`', src/fullnode.rs:954:25

---- replay_stage::test::test_replay_stage_leader_rotation_exit stdout ----
thread 'replay_stage::test::test_replay_stage_leader_rotation_exit' panicked at 'assertion failed: `(left == right)`
  left: `Some(LeaderRotation(32, 37, 8p7npd33Hd57H2kmC5TySSEtJRXZHh84qwNyX8eCnwjz))`,
 right: `Some(LeaderRotation(50, 55, EcYYbRdkYnQfowjc89bNdjpGShTyatCQHmYeVNJC1tkN))`', src/replay_stage.rs:416:9

```
It seems like the too many entires get written but since the tvu is _always_ running now, we want to keep writing entries in replay_stage. Just because my_id == scheduled leader doesn't mean rotation _just_ happened any more. Thoughts? I was going to try and track the current leader on entry to this function and only if the leader changes will I enter this block and truncate entries?